app/build.gradle

```diff
@@ -1,15 +1,19 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.3"
+    compileSdkVersion 27
 
     defaultConfig {
         applicationId "com.microsoft.o365_android_microsoft_graph_snippets"
-        minSdkVersion 16
-        targetSdkVersion 23
+        minSdkVersion 21
```

Comment:
Is it possible for the sample to work with minSdkVersion 16?  It's best to work with the lowest sdk possible without trading off too much. The lower the sdk the more devices the sample will run on.

Reply:
This is not possible, as the Android auth wrapper uses MSAL 0.2.2 which in turn uses ver. 21. Downgrading the version would lead to build issues.